patches/chrome-browser-sync-profile_sync_service_factory.cc.patch

```diff
@@ -1,13 +1,13 @@
 diff --git a/chrome/browser/sync/profile_sync_service_factory.cc b/chrome/browser/sync/profile_sync_service_factory.cc
-index 8b3b7aad21bf40af75a9d70d751873697091c31d..b6257f3d8a96d3c0e9c9b98406bcfe5182a78f59 100644
+index 8b3b7aad21bf40af75a9d70d751873697091c31d..55dceae86805b48c4f9bd304f98f60176d0b1029 100644
 --- a/chrome/browser/sync/profile_sync_service_factory.cc
 +++ b/chrome/browser/sync/profile_sync_service_factory.cc
 @@ -266,7 +266,7 @@ KeyedService* ProfileSyncServiceFactory::BuildServiceInstanceFor(
    }
  
    auto pss =
 -      std::make_unique<syncer::ProfileSyncService>(std::move(init_params));
-+      std::make_unique<syncer::BraveProfileSyncService>(std::move(init_params), profile);
++      std::make_unique<syncer::BraveProfileSyncService>(std::move(init_params), BRAVE_BUILD_SERVICE_INSTANCE_FOR);
```

Comment:
We should define `BRAVE_BUILD_SERVICE_INSTANCE_FOR` as
```
std::make_unique<syncer::BraveProfileSyncService>(std::move(init_params),
DeviceInfoSyncServiceFactory::GetForProfile(profile));
```

Reply:
Fixed at c58940a5de92ccc05b1de3fe22db003c7051ad2e